lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -96,15 +96,17 @@ class ThreadRepositoryImpl extends ThreadRepository {
       await _updateEmailCache(accountId, session.username, newCreated: networkEmailResponse.emailList);
     }
 
-    if (localEmailResponse.hasState() && getLatestChanges) {
+    if (localEmailResponse.hasState()) {
       log('ThreadRepositoryImpl::getAllEmail(): filter = ${emailFilter?.mailboxId} local has state: ${localEmailResponse.state}');
-      await _synchronizeCacheWithChanges(
-        session,
-        accountId,
-        localEmailResponse.state!,
-        propertiesCreated: propertiesCreated,
-        propertiesUpdated: propertiesUpdated
-      );
+      if (getLatestChanges) {
+        await _synchronizeCacheWithChanges(
+          session,
+          accountId,
+          localEmailResponse.state!,
+          propertiesCreated: propertiesCreated,
+          propertiesUpdated: propertiesUpdated
+        );
+      }
     } else {
```

Comment:
What is different here?

Reply:
We only update state only in:
- we don't have cache on state
- or after we successfully do `/changes`.

`getLatestChanges = false` made the state was updated whenever we switch mailbox